### ISSUE TITLE
Support for HbbTV

### ISSUE THREAD

**Issue by Soutar** [rackt/history#77]
I'm currently working on an application for [HbbTV](https://en.wikipedia.org/wiki/Hybrid_Broadcast_Broadband_TV) devices using React and React Router (and therefore this package). After upgrading to React Router v1.0.0-rc, our first history-dependent version, navigation no longer works due to the use of sessionStorage in [DOMStateStorage](https://github.com/rackt/history/blob/master/modules/DOMStateStorage.js#L9), which fails silently but prevents further execution. Unfortunately when dealing with HbbTV we often find that APIs like this are available on `window` but are unusable or crippled somehow, making polyfilling them impossible.

We're considering the following workarounds so far:

* Fork and maintain our own version of history without sessionStorage (cookies instead perhaps?)
* Fork and wrap sessionStorage calls in `try {} catch () {}`
* Extend the `Link` component and write our own `onClick` handler which grabs the `href` from the DOM node and manually navigates to it

The last two options are very quick fixes but sacrifice functionality which is not ideal. Do you have any advice for utilising this package on platforms where sessionStorage exists but throws errors when used?

**Comment by mjackson** [comment]
What mechanisms do you have for persisting state between page loads on an HbbTV? We should be able to easily create a custom state storage engine that works on those devices. For example, memory history [creates its own state storage](https://github.com/rackt/history/blob/17c1576b5b2bd6a1ec0884ede95fb8a74145dfb8/modules/createMemoryHistory.js#L6-L13), which is really just a hash.

**Comment by taion** [comment]
@Soutar Are you using hash history or browser history? If you're using hash history and not persisting things in history state, you might be able to just set `queryKey: false`.

**Comment by mjackson** [comment]
Any intention of following up here @Soutar? Please close if not.

**Comment by Soutar** [comment]
Yes, I had intended to reply here but had forgotten. Sorry @mjackson, and thanks to both you and @taion for your replies.

We discovered that the issue we were having with sessionStorage was actually device-specific and we were able to get around it. After doing some spec reading, I found that the OIPF Web Standards TV Profile (essentially part of HbbTV) mandates that "Terminals SHALL support the Web Storage API", so it's safe to say that this issue is not your responsibility.

I'm happy to close this off as history & react-router both work flawlessly on standards-compliant HbbTV devices. Thanks for providing such high quality packages :+1:
